src/components/game-board/GameBoard.tsx

```diff
@@ -43,32 +37,34 @@ function GameBoard({ className }: ChildProps): React.JSX.Element{
         navigate('/game-over');
     }, [navigate, setGame]);
 
-    const updatePlayerTurn =  useCallback(()=>{
-        setShowGameDialog(false);
-        setGame((prevGame: Game) => {
-            const currentPlayer = prevGame.activePlayer;
-            const currentPlayerIndex = prevGame.players.indexOf(currentPlayer!);
-            const nextPlayerIndex = (currentPlayerIndex + 1) % prevGame.players.length;
-            return {
-                ...prevGame,
-                totalTurns: prevGame.totalTurns - 1,
-                activePlayer: prevGame.players[nextPlayerIndex],
-                nextPlayer: prevGame.players[(nextPlayerIndex + 1) % prevGame.players.length],
+const updatePlayerInsideGameObject = useCallback((prevGame: Game) => {
+    const currentPlayer = prevGame.activePlayer;
+    const currentPlayerIndex = prevGame.players.indexOf(currentPlayer!);
+    const nextPlayerIndex = (currentPlayerIndex + 1) % prevGame.players.length;
+    return {
+        ...prevGame,
+        totalTurns: prevGame.totalTurns - 1,
+        activePlayer: prevGame.players[nextPlayerIndex],
+        nextPlayer: prevGame.players[(nextPlayerIndex + 1) % prevGame.players.length],
+
+    };
+},[]);
 
-            };
-        });
-    }, []);
+    const updatePlayerTurn = useCallback(() => {
+        setGame(updatePlayerInsideGameObject);
+    }, [updatePlayerInsideGameObject]);
 
     // useEffect to initialize the game
     useEffect(() => {
+        if(showGameDialog) {
```

Comment:
This is a side effect of action done by the GameDialog, I think we can move it to there since we have access to setGame from anywhere. We wont need the useEffect but a callBack function the will be called when the the dialog is getting closed.